component/stream/stream_cache.go

```diff
@@ -173,7 +172,13 @@ func (c *cache) getBlock(fileKey string, offset int64, fileSize int64) (*cacheBl
 			}
 			c.evictedBlock = blockKey{}
 		}
-		return newBlock, false
+
+		dataFetched := false
```

Comment:
Where is the code that fetches from the memory cache? Why do we initialize this to false so late?

Reply:
By default we only allocate block and return. So this flag is init after creating the block. Then we check if this block was earlier persisted on disk or not, if so read data from there and mark this block as already having data. Just below this line we are checking to read data from disk.